File: app/src/main/java/ru/egoncharovsky/wordstart/domain/card/LearningCard.java
package ru.egoncharovsky.wordstart.domain.card;

import ru.egoncharovsky.wordstart.domain.Entity;
import ru.egoncharovsky.wordstart.domain.word.Phrase;
import ru.egoncharovsky.wordstart.domain.word.Translation;

/**
 * Immutable
 */
public class LearningCard extends Entity {

    private final Phrase originalPhrase;

    private final Phrase translationPhrase;

    public LearningCard(Phrase originalPhrase, Phrase translationPhrase) {
        if (originalPhrase.getLanguage().equals(translationPhrase.getLanguage()))
            throw new IllegalArgumentException("Translation language must differ to original");

        this.originalPhrase = originalPhrase;
        this.translationPhrase = translationPhrase;
    }

    public LearningCard reverse() {
        return new LearningCard(translationPhrase, originalPhrase);
    }

    public Phrase getOriginalPhrase() {
        return originalPhrase;
    }

    public Phrase getTranslationPhrase() {
        return translationPhrase;
    }

    public boolean containedInTranslation(Translation translation) {
        return translation.getOriginalPhrase().equals(originalPhrase)
                && translation.variantsContainsWord(translationPhrase);
    }
}

File: app/src/main/kotlin/ru/egoncharovsky/wordstart/domain/card/LearningCard.kt
package ru.egoncharovsky.wordstart.domain.card

import ru.egoncharovsky.wordstart.domain.Identifiable
import ru.egoncharovsky.wordstart.domain.word.Phrase
import ru.egoncharovsky.wordstart.domain.word.Translation

data class LearningCard(
        var id: Long?,
        val front: Phrase,
        val back: Phrase
) : Identifiable<Long> {
    override fun id(): Long? = id

    fun containedIn(translation: Translation) =
            translation.originalPhrase == front && translation.variantsContainsWord(back)
}